ciri/api/api.c

```diff
@@ -434,15 +434,72 @@ retcode_t iota_api_were_addresses_spent_from(
 retcode_t iota_api_check_consistency(iota_api_t const *const api,
                                      check_consistency_req_t const *const req,
                                      check_consistency_res_t *const res) {
+  retcode_t ret = RC_OK;
+  hash243_queue_entry_t *iter = NULL;
+  bundle_transactions_t *bundle = NULL;
+  bundle_status_t bundle_status = BUNDLE_NOT_INITIALIZED;
+  exit_prob_transaction_validator_t walker_validator;
+  DECLARE_PACK_SINGLE_TX(tx, txp, pack);
+
   if (api == NULL || req == NULL || res == NULL) {
     return RC_NULL_PARAM;
   }
 
+  res->state = false;
+
   if (invalid_subtangle_status(api)) {
     return RC_API_INVALID_SUBTANGLE_STATUS;
   }
 
-  return RC_OK;
+  CDL_FOREACH(req->tails, iter) {
+    bundle_transactions_new(&bundle);
+    hash_pack_reset(&pack);
+    if ((ret = iota_tangle_transaction_load(&api->consensus->tangle,
+                                            TRANSACTION_FIELD_HASH, iter->hash,
+                                            &pack)) != RC_OK) {
+    } else if (pack.num_loaded == 0) {
+      ret = RC_API_TAIL_MISSING;
```

Comment:
why won't you set that in res?

Reply:
For IRI-compatibility, some errors are returned with an error response, some are included in the `info` field of the `checkConsistency` response...

Reply:
Because some are errors due to non-respect of the API contract, some are just that the tails are not consistent and `info` tells why